packages/scene-composer/src/components/panels/SceneHierarchyPanel/SceneHierarchyDataProvider.tsx

```diff
@@ -192,50 +192,53 @@ const SceneHierarchyDataProvider: FC<SceneHierarchyDataProviderProps> = ({ selec
-            {
-              position: worldPosition,
-              rotation: worldRotation,
-              scale: worldScale,
-            },
-            newParentObject,
-          );
-        }
-        // remove child ref from parent
-        if (!oldParentRef) {
-          const newRoots = document.rootNodeRefs.filter((ref) => ref !== objectRef);
-          updateDocumentInternal({ rootNodeRefs: newRoots });
-        } else {
-          updateSceneNodeInternal(oldParentRef, { childRefs: oldParentChildren });
-        }
-        // Create updates to the moving object
-        const partial: RecursivePartial<ISceneNodeInternal> = { parentRef: newParentRef };
-        if (maintainedTransform) {
-          // Update the node position to remain in its world space
-          partial.transform = {
-            position: maintainedTransform.position.toArray(),
-            rotation: maintainedTransform.rotation.toVector3().toArray(),
-            scale: maintainedTransform.scale.toArray(),
-          };
-        }
+      const objectToMoveRef = originalObject?.ref as string;
+      const oldParentRef = originalObject?.parentRef as string;
+      const oldParent = getSceneNodeByRef(oldParentRef);
+      const oldParentChildren = oldParent?.childRefs.filter((child) => child !== objectToMoveRef);
+      let maintainedTransform: any = null;
+      const newParentObject = getObject3DBySceneNodeRef(newParentRef);
+      const newParent = getSceneNodeByRef(newParentRef);
+
+      if (originalObject3D) {
+        const worldPosition = originalObject3D.getWorldPosition(new Vector3());
+        const worldRotation = new Euler().setFromQuaternion(originalObject3D.getWorldQuaternion(new Quaternion()));
+        const worldScale = originalObject3D.getWorldScale(new Vector3());
+        maintainedTransform = getFinalTransform(
+          {
+            position: worldPosition,
+            rotation: worldRotation,
+            scale: worldScale,
+          },
+          newParentObject,
+        );
+      }
+      // remove child ref from parent
+      if (!oldParentRef) {
+        const newRoots = document.rootNodeRefs.filter((ref) => ref !== objectRef);
+        updateDocumentInternal({ rootNodeRefs: newRoots });
+      } else {
+        updateSceneNodeInternal(oldParentRef, { childRefs: oldParentChildren });
+      }
+      // Create updates to the moving object
+      const partial: RecursivePartial<ISceneNodeInternal> = { parentRef: newParentRef };
+      if (maintainedTransform) {
+        // Update the node position to remain in its world space
+        partial.transform = {
+          position: maintainedTransform.position.toArray(),
+          rotation: maintainedTransform.rotation.toVector3().toArray(),
+          scale: maintainedTransform.scale.toArray(),
+        };
+      }
+      // update node to have new parent
+      if (!newParentRef) {
+        updateSceneNodeInternal(objectToMoveRef, { parentRef: newParentRef });
```

Comment:
when the node is dropped at root, we should also add it to `rootNodeRefs`

Reply:
There's no need to modify the root as it gets updated with the re-renders. Validated that the updates were persistent in refresh via local console testing.